src/spritefeat.asm

```diff
@@ -0,0 +1,336 @@
+;
+; Features using sprites to draw information
+;
+
+
+; Hook Gamemode 7 to upload tiles after loading if needed
+org $828B26
+    JSL update_sprite_tiles_loading
+
+; Hook sprite uploads in main game loop to get a chance to upload sprites first
+; Runs in Gamemode 8
+org $828B98
+    JSL update_sprite_features
+
+
+
+;
+; Add hitbox graphic to free sprite VRAM slot
+;
+org $9ADAE0
+db $FF, $FF, $80, $80, $80, $80, $80, $80, $80, $80, $80, $80, $80, $80, $80, $80
+db $FF, $FF, $80, $80, $80, $80, $80, $80, $80, $80, $80, $80, $80, $80, $80, $80 
+
+
+org $F0E000
+print pc, " spritefeat start"
+; This runs every frame before any other sprites are drawn, this is needed so we can get priority over everything else
+update_sprite_features:
```

Comment:
By my count, by default (new features off) this method adds 90 clock cycles per gameplay frame.  Adding in 8 cycles for the new check in the artificial lag check and we've got 98 cycles... that's a lot!

We should decrease the default artificial lag by one to get back 80 cycles.  I'd also recommend optimizing this method a bit.  At the beginning of the two JSR methods is a check, when false we RTS.  We should be doing that here and only JSR if we need to (saving a JSR and RTS each for 24 cycle savings total).  That would make the total CPU hit almost even.